staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -349,6 +664,36 @@ func (a *Authenticator) AuthenticateToken(token string) (user.Info, bool, error)
 	return info, true, nil
 }
 
+// getClaimJWT gets a distributed claim JWT from url, using the supplied access
+// token as bearer token.  If the access token is "", the authorization header
+// will not be set.
+// TODO: Allow passing in JSON hints to the IDP.
+func getClaimJWT(t *http.Transport, url, accessToken string) (string, error) {
+	client := &http.Client{Transport: t, Timeout: 30 * time.Second}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// TODO: Allow passing request body with configurable information.
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", fmt.Errorf("while calling %v: %v", url, err)
+	}
+	if accessToken != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", accessToken))
+	}
+	req = req.WithContext(ctx)
+	response, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	responseBytes, err := ioutil.ReadAll(response.Body)
+	defer response.Body.Close()
```

Comment:
Move this before ReadAll. It won't make Reads fail

Reply:
Is there any specific reason to prefer that idiom?

Reply:
It follows the general pattern of

```
r, err := allocateResource()
if err != nil { return err }
defer cleanupResource(r)
```

If you keep it as is, every reader will think there's some non-obvious semantics and spend time trying to understand the reasoning behind it.

Reply:
Oh, I see. I was thinking more along the lines of "the first use of `response.Body` is in line `N`".  So it made sense to defer close in line `N+1`. 

Whereas, in fact, the resource was allocated at `... := client.Do(...)`.